Packs/FortiGate/Integrations/FortiGate/FortiGate.py

```diff
@@ -1,93 +1,2131 @@
+        args["mask"] = "255.255.255.255"
 
-@logger
-def http_request(method, url_suffix, params={}, data=None):
+    # Identify fully set, partially set, and mixed groups
+    fully_set_groups = []
+    partially_set_groups = []
+
+    for group, count in group_to_arg_counts.items():
+        if count >= len(group_to_arg_names[group]):
+            fully_set_groups.append(group)
+        elif 0 < count < len(group_to_arg_names[group]):
+            partially_set_groups.append(group)
+
+    mixed_groups = fully_set_groups + partially_set_groups
+
+    # If arguments are from more than one group, raise an error
+    if len(mixed_groups) > 1:
+        mixed_groups_str = ", ".join(mixed_groups)
+        raise DemistoException(f"Arguments must only come from one group. Mixed groups: {mixed_groups_str}")
+
+    # If no group is fully set, raise an error
+    if not fully_set_groups:
+        if partially_set_groups:
+            raise DemistoException(
+                f"Missing arguments for the group {partially_set_groups[0]}, "
+                f"please provide all: {group_to_arg_names[partially_set_groups[0]]}"
+            )
+
+        raise DemistoException(
+            "No group of arguments was fully set. "
+            f"Please provide arguments from one of the following groups: {list(group_to_arg_names)}"
+        )
+
+    gui_to_api = ADDRESS6_GUI_TO_API_TYPE if include_ipv6 else ADDRESS_GUI_TO_API_TYPE
+
+    return gui_to_api[fully_set_groups[0]]
+
+
+def get_service_type(args: dict[str, Any]) -> str:
+    """Identifies the protocol type of argument based on the arguments.
 
-    res = SESSION.request(
-        method,
-        BASE_URL + url_suffix,
-        verify=USE_SSL,
-        params=params,
-        data=data
+    Args:
+        args (dict[str, Any]): The arguments to check.
+
+    Returns:
+        str: The type of argument group provided.
+            Possible values are: "tcp", "udp", "sctp", "ip", "icmp", and "icmp6".
+
+    Raises:
+        DemistoException:
+            - If arguments from more than one group are provided.
+            - If not all arguments from a single group are provided.
+            - If no group of arguments is fully set.
+    """
+    protocol_type_to_arg_names = {
+        TCP_UDP_SCTP: ["start_ip", "end_ip", "fqdn", "tcpRange", "udpRange", "sctpRange"],
+        IP: ["ip_protocol"],
+        f"{ICMP}/{ICMP6}": ["icmp_version", "icmp_type", "icmp_code"],
+    }
+
+    # Count the number of not None arguments for each protocol type
+    protocol_type_to_arg_counts = {
+        protocol_type: sum(1 for arg in arg_names if args.get(arg) is not None)
+        for protocol_type, arg_names in protocol_type_to_arg_names.items()
+    }
+
+    # Identify fully set, partially set, and mixed groups
+    fully_set_protocol_types = []
+    partially_set_protocol_types = []
+
+    for protocol_type, count in protocol_type_to_arg_counts.items():
+        if count >= len(protocol_type_to_arg_names[protocol_type]):
```

Comment:
Not really, it is just some thing I usually do as a safety net.
I guess because all the variables are within the same scope I'll change it to only `eq`.